### ISSUE TITLE
Saved ImageJ test image produces a tif that crashes tifffile.imread()

### ISSUE THREAD

**Issue by marimar128** [cgohlke/tifffile#6]
Steps to reproduce:
Open ImageJ 1.52v
Run this ImageJ macro:

```
run("HeLa Cells (1.3M, 48-bit RGB)");
//run("Channels Tool...");
Stack.setDisplayMode("grayscale");
saveAs("Tiff", "C:/Users/Admin/Desktop/tifffile_debugging/hela-cells.tif");
```
(If you omit the line `Stack.setDisplayMode("grayscale");`, the tif opens as expected)

Now run the following python code:
```
import tifffile
print(tifffile.__version__)
data = tifffile.imread('hela-cells.tif')
```
Which yields the following output:
```
2020.5.7
Traceback (most recent call last):
  File "C:\Users\Admin\Desktop\tifffile_debugging\test.py", line 12, in <module>
    data = tifffile.imread('hela-cells.tif')
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python38\lib\site-packages\tifffile\tifffile.py", line 589, in imread
    return tif.asarray(**kwargs)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python38\lib\site-packages\tifffile\tifffile.py", line 2306, in asarray
    result = self.filehandle.read_array(
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python38\lib\site-packages\tifffile\tifffile.py", line 6402, in read_array
    raise ValueError(f'failed to read {size} bytes')
ValueError: failed to read 6193152 bytes
>>> 
```
Before this line, `shape` is [3]:
https://github.com/cgohlke/tifffile/blob/e1902a9a1574223d82e06b97788f9a0465015359/tifffile/tifffile.py#L2656
After this line, `shape` is [3, 512, 672, 3], which doesn't match the file size.

Thanks for making tifffile, it's a fantastic tool that I've always been grateful for!

**Comment by cgohlke** [comment]
Thank you. Should be fixed in v2020.5.11.

**Comment by marimar128** [comment]
That fixed it for me! Thank you, you're awesome :)
